core/client/db/src/lib.rs

```diff
@@ -104,10 +104,10 @@ pub fn new_client<E, S, Block>(
 	block_execution_strategy: ExecutionStrategy,
 	api_execution_strategy: ExecutionStrategy,
 ) -> Result<client::Client<Backend<Block>, client::LocalCallExecutor<Backend<Block>, E>, Block>, client::error::Error>
-	where
-		Block: BlockT,
-		E: CodeExecutor<Blake2Hasher> + RuntimeInfo,
-		S: BuildStorage,
+where
+	Block: BlockT<Hash=H256>,
```

Comment:
Where is this requirement coming from? It is unfortunate to see it propagating all the way up.

Reply:
It's from the previous-implicit requirement that `H::Out == Block::Hash`. As we have been using `Blake2Hasher` specialization, that means in many places we're practically already limited to `H256`. So I don't think we've lost generalization over this.